.github/workflows/tests.yml

```diff
@@ -11,7 +11,7 @@ on:
 jobs:
 
   static-code-analysis:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```

Comment:
Ubuntu 20-04 is deprecated since the beginning of the month: https://github.com/actions/runner-images/issues/11101

Please update to `ubuntu-24.04` - or better yet: revert back to `ubuntu-latest` (it's not quite clear to me why you pinned/downgraded it)